auctions/english-auction/tests/utils/mod.rs

```diff
@@ -0,0 +1,442 @@
+        let reserve_price = 10;
+        let duration = 10;
+
+        (sell_amount, initial_price, reserve_price, duration)
+    }
+
+    pub async fn nft_asset(contract_id: ContractId, token_id: u64) -> AuctionAsset {
+        let token = NFTAsset {
+            contract_id,
+            token_id,
+        };
+
+        AuctionAsset::NFTAsset(token)
+    }
+
+    pub async fn setup() -> (
+        Metadata,
+        Metadata,
+        Metadata,
+        Metadata,
+        ContractId,
+        ContractId,
+        ContractId,
+        ContractId,
+        ContractId,
+    ) {
+        let num_wallets = 4;
+        let coins_per_wallet = 1;
+        let coin_amount = 1000000;
+        let config = Config {
+            manual_blocks_enabled: true, // Necessary so the `produce_blocks` API can be used locally
+            ..Config::local_node()
+        };
+        let mut wallets = launch_custom_provider_and_get_wallets(
+            WalletsConfig::new(Some(num_wallets), Some(coins_per_wallet), Some(coin_amount)),
+            Some(config),
+        )
+        .await;
+
+        // Get the wallets from that provider
+        let wallet1 = wallets.pop().unwrap();
+        let wallet2 = wallets.pop().unwrap();
+        let wallet3 = wallets.pop().unwrap();
+        let wallet4 = wallets.pop().unwrap();
+
+        let auction_id = Contract::deploy(
+            "./out/debug/english-auction.bin",
+            &wallet1,
+            TxParameters::default(),
+            StorageConfiguration::with_storage_path(Some(
+                "./out/debug/english-auction-storage_slots.json".to_string(),
+            )),
+        )
+        .await
+        .unwrap();
+
+        let sell_asset_id = Contract::deploy(
+            "./tests/artifacts/asset/out/debug/asset.bin",
+            &wallet1,
+            TxParameters::default(),
+            StorageConfiguration::default(),
+        )
+        .await
+        .unwrap();
+
+        let sell_nft_id = Contract::deploy(
+            "../../NFT/out/debug/NFT.bin",
+            &wallet1,
+            TxParameters::default(),
+            StorageConfiguration::with_storage_path(Some(
+                "../../NFT/out/debug/NFT-storage_slots.json".to_string(),
+            )),
+        )
+        .await
+        .unwrap();
+
+        let deploy_wallet = Metadata {
+            asset: MyAsset::new(sell_asset_id.to_string(), wallet1.clone()),
+            auction: EnglishAuction::new(auction_id.to_string(), wallet1.clone()),
+            nft: Nft::new(sell_nft_id.to_string(), wallet1.clone()),
+            wallet: wallet1.clone(),
+        };
+
+        let seller = Metadata {
+            asset: MyAsset::new(sell_asset_id.to_string(), wallet2.clone()),
+            auction: EnglishAuction::new(auction_id.to_string(), wallet2.clone()),
+            nft: Nft::new(sell_nft_id.to_string(), wallet2.clone()),
+            wallet: wallet2.clone(),
+        };
+
+        let rng = &mut StdRng::seed_from_u64(2322u64);
```

Comment:
I was wondering why `rand` is added as a dependency. In this case a seed is being used to generate a salt so the salt should always be the same. Is there a reason for the magic number or is it purely magic? Can that be added as comments?

I generally advise against the usage of randomness in tests because of the boundary conditions and unexpected failures that may happen. I'd prefer a hardcoded value however the seed will constrain the randomness so it comes down to whether we want that additional dependency and compilation time.

Reply:
I'm following the [Rust SDK Book](https://fuellabs.github.io/fuels-rs/v0.26.0/getting-started/contracts.html) here.